src/libmongoc/tests/test-mongoc-bulk.c

```diff
@@ -3773,6 +3777,14 @@ test_bulk_max_msg_size (void)
                        strlen ("$clusterTime") + 2;
    }
 
+   /* Versioned API may be appended */
+   if (client->api) {
+      /* TODO: Figure out why 7 works. */
```

Comment:
Hmm, we would expect to just subtract `2`, right? For the null terminators of the two strings? Was it just guess and check that made 7 work?

Reply:
Yes, careful trial-and-error revealed 7 as the magic number. All others append 2, so I'm not entirely sure what's going on. I'll step-debug this and then we'll see.

Reply:
Through debugging I was able to confirm that the `bson_append_utf8` call in [`_mongoc_cmd_append_server_api`](https://github.com/mongodb/mongo-c-driver/blob/15cbc7865a513c09859fd4c88ba296b9b01139e2/src/libmongoc/src/mongoc/mongoc-cmd.c#L1168) indeed adds 18 bytes to the BSON. 10 for the "apiVersion" string, 1 for the actual string, plus 7 characters that do something. I've removed the comment, and maybe someone with knowledge of this could see if we can find a better way to do this.